hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/HiveSyncContext.java

```diff
@@ -75,7 +78,12 @@ private HiveSyncContext(Properties props, HiveConf hiveConf) {
   public HiveSyncTool hiveSyncTool() {
     HiveSyncMode syncMode = HiveSyncMode.of(props.getProperty(HIVE_SYNC_MODE.key()));
     if (syncMode == HiveSyncMode.GLUE) {
-      return new AwsGlueCatalogSyncTool(props, hiveConf);
+      if (ReflectionUtils.hasConstructor(AWS_GLUE_CATALOG_SYNC_TOOL_CLASS,
+          new Class<?>[] {Properties.class, org.apache.hadoop.conf.Configuration.class})) {
```

Comment:
Do we need the if check? We can not fallback to hive sync tool if user expects GLUE.

Reply:
Thanks for pointing out the unnecessary if condition. I have fixed it in the latest commit.